src/Service/PageRawQuery.php

```diff
@@ -68,4 +69,48 @@ public function update(int $id, array $fields)
 
         $stmt->execute($fields);
     }
+
+    public function getPathWithShortcode(string $shortcode): ?string
+    {
+        $pctCount = "
+            SELECT COUNT(pct.id)
+            FROM `page_content_text` pct
+            WHERE pct.page_revision_id = pr.id
+            AND pct.type = 'wysiwyg'
+            AND pct.text LIKE :shortcode
+        ";
+
+        $pcrOneColumnOr = "pcr.layout = 'one_column' AND pcr.column_1 LIKE :shortcode";
+        $pcrTwoColumnsOr = "pcr.layout NOT IN ('one_column', 'three_column') AND (pcr.column_1 LIKE :shortcode OR pcr.column_2 LIKE :shortcode)";
+        $pcrThreeColumnsOr = "pcr.layout = 'three_column' AND (pcr.column_1 LIKE :shortcode OR pcr.column_2 LIKE :shortcode OR pcr.column_3 LIKE :shortcode)";
```

Comment:
Here instead of just looking at all 3 columns, I'm only looking at the relevant columns because depending on the layout, some columns are not output even if they have content.